### ISSUE TITLE
Vertical spacing with line breaks and above-line text

### ISSUE THREAD

**Issue by rarty** [gregorio-project/gregorio#533]
In a score with manual line breaks and above-line text, the vertical spacing between the bottom of the 'lyric' line and the top of the above-line text of the next line can be inconsistent. It seems that in this case, the distance `abovelinestextheight` is added to all the lines with above-line text, except for the first line past the manual line break.

(Here `abovelinestextheight` is set to 1 cm to emphasize the difference)

    (c4)Ma(f)gní(gh)fi(h)cat(h.) (::Z)
    Ma(f)gní(gh)fi(h)cat([alt:Above-line text]h.) (::Z)
    Ma(f)gní(gh)fi(h)cat([alt:Above-line text]h.) (::)

![screen shot 2015-06-30 at 9 27 13 pm](https://cloud.githubusercontent.com/assets/6454162/8446365/d79f0794-1f6e-11e5-9cb5-9c90bce6e8da.png)

In the case of a score with no manual line breaks, it seems `abovelinestextheight` is not added at all.

```
(c4)Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma([alt:Above-line text]f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma([alt:Above-line text]f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma(f)gní(gh)fi(h)cat(h.) (::)
Ma([alt:Above-line text]f)gní(gh)fi(h)cat(h.) (::)
```
![screen shot 2015-06-30 at 9 39 39 pm](https://cloud.githubusercontent.com/assets/6454162/8446471/8bdb7c0a-1f70-11e5-9bfe-d9b5ec7c8803.png)

I'm not sure what the correct behavior should be (with or without the added space).

I have noticed this for some time, so I don't think it is a behavior introduced recently, but the testing I did just now was using build `54a3036` from the develop branch.

**Comment by henryso** [comment]
I noticed, while I was adding the variable line height implementation, that gregorio was only setting spaces below the line at the start of a score.  Though this didn't seem orthogonal with the manual line breaks, not knowing any better, I just assumed that was the correct behavior.  What you see appears to be consistent with what I saw.  As for me, I would think that the space should be added regardless of whether there is a manual line break, but I don't know if this was done intentionally for some reason or is just an oversight.

As @rpspringuel is intimately familiar with the spacing code, he might be the best person to address this.

**Comment by henryso** [comment]
Thinking about this more, I'm beginning to feel that:

1. When GregorioTeX is in uniform line height expansion mode, the above lines text space and the space for anything else above and below the line should be added uniformly.  This means that the values computed by gregorio (the executable) for top/bottom height as well as for translation space and above lines text space should all be added to all lines so that the lines are uniform.
2. When GregorioTeX is in variable line height expansion mode, the Lua algorithm needs to determine whether there is above lines text or translation space for the given line and add the spaces as necessary, per line.

What does everyone think?  Does this sound correct?

**Comment by eroux** [comment]
Agreed!

**Comment by eschwab** [comment]
I most certainly agree. Since most of the hard work was done for #59 it makes sense to have all the lines taken into account in `variable` mode.

**Comment by rpspringuel** [comment]
I agree as well.  Frankly, I never realized this was a problem because my work on the spacing code had more to do with how spaces were stored and used rather than in when they were calculated and/or used.

**Comment by henryso** [comment]
The fix in #534 is ready for review.
